docs/usage/webhook/deploy.rst

```diff
@@ -0,0 +1,117 @@
+.. _usage_ffmpeg_deploy:
+
+In this tutorial we will deploy a dataflow(ffmpeg dataflow) which converts a video to gif over an http server.We'll also
+see how to deploy the same http server in a docker container.Finally in :ref:`usage_ffmpeg_deploy_serve`
+we'll setup another http server which waits on gtihub webhooks to rebuilt and deploy the ffmpeg dataflow.
+
+Deploying on http server
+========================
+
+.. note::
+
+    All the code for this example is located under the
+    `examples/ffmpeg <https://github.com/intel/dffml/blob/master/examples/ffmpeg/>`_
+    directory of the DFFML source code.Rest of the tutorial is ran from the root of
+    this folder.
+
+Write operations and definitions to convert videos files to gif by calling
+``ffmpeg`` (Make sure you `download and install <https://www.ffmpeg.org/download.html>`_ it.).
+The operation accepts bytes (of the video), converts it to gif and outputs it as bytes.
+
+**ffmpeg/operations.py**
+
+.. literalinclude:: /../examples/ffmpeg/ffmpeg/operations.py
+
+Dataflow and Config files
+-------------------------
+
+**ffmpeg/dataflow.py**
+
+.. literalinclude:: /../examples/ffmpeg/ffmpeg/dataflow.py
+
+.. code-block:: console
+
+    $ mkdir -p deploy/mc/http deploy/df
+    $ dffml service dev export -config yaml ffmpeg.dataflow:DATAFLOW > deploy/df/ffmpeg.yaml
+
+Create the config file for the http server
+in ``deploy/mc/http/ffmpeg.yaml``
+
+.. code-block:: console
+
+    $ cat > ./deploy/mc/http/ffmpeg.yaml <<EOF
+    path: /ffmpeg
+    input_mode: bytes:Input_file
+    output_mode: bytes:post_input.Output_file
+    EOF
+
+- ``input_mode``
+    - ``bytes:Input_file``
+        - we want the input from the request to be treated as bytes with definition ``Input_file``.
+- ``output_mode``
+    - ``bytes:post_input.Output_file``
+        - we want the response to be bytes taken from  ``results["post_input"]["Output_file"]``, where ``results`` is the output of the dataflow.
+
+For more details see `HttpChannelConfig <../../plugins/service/http/dataflow.html#HttpChannelConfig>`__ .
+
+.. _usage_ffmpeg_deploy_serve:
+
+Serving the DataFlow
+--------------------
+
+Serving the dataflow on port 8080
+
+.. code-block:: console
+
+    $ dffml service http server -insecure -mc-config deploy -port 8080
+
+.. warning::
+
+    The ``-insecure`` flag is only being used here to speed up this
+    tutorial. See documentation on HTTP API
+    :doc:`/plugins/service/http/security` for more information.
+
```

Comment:
We should probably tell people to `curl` down the video that you've added to the GitHub repo. Or use their own if they have one.